### PR TITLE
Move from CircleCI to GitHub Actions

### LINKED ISSUE #1795: Move from CCI to GH CI
[...] move them over at your leisure.

_Originally posted by @ryjones in https://github.com/open-quantum-safe/liboqs/issues/1789#issuecomment-2111144389_

### PR DESCRIPTION
**Pull request by SWilson4**
This PR moves our remaining CircleCI workflows over to GitHub Actions. I've left it in draft for now as it's based on the branch for #1848.

I have tried to duplicate the testing coverage from CircleCI without additions or removals. This PR isn't intended to overhaul or evaluate our test coverage, it's just meant to move it to a different platform.

Three items of note:

(1) The CCI [`bionic-i386`](https://github.com/open-quantum-safe/liboqs/blob/60cd3d83c92ae2679df3460955921570a6e0f05d/.circleci/config.yml#L297) could not be ported over, as the GitHub `checkout` action requires a newer version of `node` and can't run in the older container (at least not without some finicky work). Bionic is no longer supported anyhow, so it's probably for the best that we're no longer using it for testing. However, the multiarch images that we'd been using don't have options for i386 past Bionic, so there is no easy way to continue testing on i386. We don't list Linux / x86 as a [supported platform](https://github.com/open-quantum-safe/liboqs/blob/60cd3d83c92ae2679df3460955921570a6e0f05d/PLATFORMS.md), so I've opted to simply remove CI support for it.

(2) The CCI `focal-clang-15` job had `-DOQS_OPT_TARGET=skylake` set. It didn't configure `-DOQS_DIST_BUILD=OFF`, however, so the opt target value was ignored. I have omitted the opt target variable in GitHub CI to reflect what was actually being tested.

(3) Finally, @ryjones helped me add a CircleCI API token to the liboqs environment so that GitHub can trigger CircleCI for downstream projects. These triggers are contained in the new `commit-to-main.yml` file. We may want to revise them for completeness, but for now I have left them as they were on CCI.

Fixes #1795.

* [ ] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
* [ ] Does this PR change the list of algorithms available -- either adding, removing, or renaming? Does this PR otherwise change an API? (If so, PRs in fully supported downstream projects dependent on these, i.e., [oqs-provider](https://github.com/open-quantum-safe/oqs-provider) will also need to be ready for review and merge by the time this is merged.)

### CODE REVIEW

File: .github/workflows/commit-to-main.yml
Comment (baentsch) [inline .github/workflows/commit-to-main.yml:24]: Doesn't look necessary to keep this one

File: .github/workflows/commit-to-main.yml
Comment (baentsch) [inline .github/workflows/commit-to-main.yml:44]: When to add/change this to v9?

File: .github/workflows/commit-to-main.yml
Comment (baentsch) [inline .github/workflows/commit-to-main.yml:75]: Doesn't look as necessary as triggering liboqs-rust, does it?

File: .github/workflows/unix.yml
Comment (baentsch) [inline .github/workflows/unix.yml:88]: Why did the 'auto' (# of processes started) param get dropped?

**Review by baentsch — APPROVED** [review]

File: .github/workflows/commit-to-main.yml
Comment (dstebila) [reply .github/workflows/commit-to-main.yml:44]: The v9 branch doesn't yet work; the [current PR from @geedo0](https://github.com/open-quantum-safe/openssh/pull/160) gets the basic test suite running but still doesn't have full PQ or hybrid support.  I think we can merge this PR with v8, and then when v9 is ready, we can come back and update the liboqs CI.

**Review by dstebila — APPROVED** [review]

File: .github/workflows/unix.yml
Comment (SWilson4) [reply .github/workflows/unix.yml:88]: It's still in effect; I just factored it out into the `python3 -m pytest ...` command below. It was used almost universally, so I figured that there was no point in duplicating it across all jobs.

File: .github/workflows/commit-to-main.yml
Comment (SWilson4) [reply .github/workflows/commit-to-main.yml:75]: Or liboqs-go... My intent with this PR was just to duplicate existing CCI functionality as closely as possible to make sure it worked on GH and make any improvements in a follow-up PR. But I could do both in one shot, if that would be preferable from a review standpoint.

File: .github/workflows/commit-to-main.yml
Comment (SWilson4) [reply .github/workflows/commit-to-main.yml:24]: Same comment as below re liboqs-rust applies here.

File: .github/workflows/commit-to-main.yml
Comment (SWilson4) [reply .github/workflows/commit-to-main.yml:75]: @baentsch I'll interpret the thumbs-up as an endorsement of a follow-up PR to address #1783 and merge this as is.

**Comment by SWilson4** [comment]
The [downstream trigger](https://github.com/open-quantum-safe/liboqs/actions/runs/10080975646) failed after the merge. I can trigger downstream pipelines locally with the same API token that should be in CI, so I think this is a GitHub variables config error. Will follow up with @ryjones later today.

**Pull request merged**
